### PR TITLE
Update ResourceManager SDK to 1.2.0-preview version. 

### PR DESCRIPTION
**Pull request by TianoMS**
This checklist is used to make sure that common issues in a pull request are covered by the creator. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).

Below in **Overall Changes**, check off the boxes that apply to your PR. Within each of the categories that you select, make sure that you can check off **all** of the boxes.
## Overall Changes
- [x] [**MANDATORY** - General changes](#general)
- [x] [**MANDATORY** - Add/remove/edit test(s)](#tests)
### General
- [x] Title of the PR is clear and informative
- [ ] There are a small number of commits that each have an informative message
- [ ] If it applies, references the bug/issue that the PR fixes
- [x] All files have the Microsoft copyright header
- [x] Cmdlets refer to management libraries through nuget references - no dlls are checked in
- [x] The PR does not introduce breaking changes (unless a major version change occurs in the assembly and module)
### Tests
- [x] PR includes test coverage for the included changes
- [x] Tests must use xunit, and should either use Moq to mock management client calls, or use the scenario test framework
- [x] PowerShell scripts used in tests must not use hard-coded values for location
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for existing resources
- [x] Tests should not use App.config files for settings
- [x] Tests should use the built-in PowerShell functions for generating random names when unique names are necessary - this will store names in the test recording
- [x] Tests should use Start-Sleep to pause rather than Thread.Sleep

### CODE REVIEW

**Comment by TianoMS** [comment]
@markcowl My change upgrades to Microsoft.Azure.Management.ResourceManager.1.1.5-preview. I did it only for the Resources solution. Most of the other projects are still using 1.1.1-preview version. Let me know if you see a problem in it.

**Comment by TianoMS** [comment]
On demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1112/

**Comment by vivsriaus** [comment]
@markcowl @hovsepm Adding on @TianoMS comment above regarding different versions of the same assembly. I recall there was an issue with the way powershell loaded assemblies that differed in just the maintenance version (1.1.1 vs 1.1.5). Since some projects reference 1.1.1 and ours (with this PR) reference 1.1.5, we want to ensure that this doesn't pose any problems for our cmdlets. i.e. - we don't have to force the powershell to load the 1.1.5 assembly via Add-Type to use it in our cmdlet.

File: src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj [hunk 1]
```diff
@@ -116,7 +116,7 @@
     </Reference>
     <Reference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.TestFramework.1.3.4-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.TestFramework.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.TestFramework.1.3.6-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.TestFramework.dll</HintPath>
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj]: Was this intentional?

**Comment by vivsriaus** [comment]
@TianoMS Since this is an api-version bump, why didn't we update the assembly's minor version instead (1.2.0 instead of 1.1.5)?

File: src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj (hunk 1 above)
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj]: This is about another issue. I think it's already fixed. Should not be a problem.

**Comment by TianoMS** [comment]
@vivsriaus I have another change in SDK (the parenthesis in resource group names) recently which updated the version to 1.1.6-preview. It's not published yet. Maybe we can update the version to 1.2.0 there.

**Comment by vivsriaus** [comment]
Yes, let's please do that @TianoMS. api-version bumps definitely warrants at least a minor version update. We should be fine keeping the powershell changes as-is, since the implementation is internal.

**Comment by cormacpayne** [comment]
@TianoMS Hey Tian, your on-demand build failed because of a bug in a test that we merged a fix for yesterday evening, so if you pull from dev and rebase, that should fix the problem.

**Comment by markcowl** [comment]
LGTM once the on-demand build passes.

on demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1122/

**Comment by TianoMS** [comment]
@markcowl Please hold on with the merge. We have a newer version of SDK coming out. I'll push one more commit to upgrade to that version (1.2.0) directly.

**Comment by TianoMS** [comment]
This is now ready to merge. On demand job:
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1126/

File: src/ResourceManager/Resources/Commands.Resources.Test/Providers/GetAzureProviderCmdletTests.cs [hunk 2]
```diff
@@ -153,7 +153,7 @@ public void GetsResourceProviderTests()
             };
             var pagableLocations = new Page<Location>();
             pagableLocations.SetItemValue<Location>(locationList);
-            var locationsResult = new AzureOperationResponse<IPage<Location>>()
+            var locationsResult = new AzureOperationResponse<IEnumerable<Location>>()
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources.Test/Providers/GetAzureProviderCmdletTests.cs]: This is a breaking change - why is this operation no longer pageable?

**Comment by markcowl** [comment]
@TianoMS One question about the removal of a pageable operation

File: src/ResourceManager/Resources/Commands.Resources.Test/Providers/GetAzureProviderCmdletTests.cs (hunk 2 above)
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.Resources.Test/Providers/GetAzureProviderCmdletTests.cs]: The change really happens here: https://github.com/Azure/azure-rest-api-specs/commit/5129caaf3816b18fa0e32f3e8e5ece7b9e70f6e6. 

At the bottom, it sets "nextLinkName" to null. This makes the generated SDK not pageable. ISubscriptionsOperations.ListLocationsWithHttpMessagesAsync now returns IEnumerable. 

It's not a breaking change here. It's a breaking change in .NET SDK. I can sync up with Vivek on that change in swagger spec. 

This should be a

**Comment by TianoMS** [comment]
@markcowl Addressed.

File: src/ResourceManager/Resources/Commands.Resources.Test/Providers/GetAzureProviderCmdletTests.cs (hunk 2 above)
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.Resources.Test/Providers/GetAzureProviderCmdletTests.cs]: @TianoMS @markcowl That is correct. The list subscription locations was never pageable to begin with, it was incorrectly added. The above change to swagger spec @TianoMS referenced fixed the issue.

Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources.Test/Providers/GetAzureProviderCmdletTests.cs]: As long as it's not a breaking change here, it's OK, but IMO, you want to fix this in the sdk, you don't want to go from pageange to not pageable back to pageable

**Pull request merged**